crates/trie/trie/src/node_iter.rs

```diff
@@ -282,7 +288,23 @@ where
                         continue
                     }
 
-                    self.current_hashed_entry = self.seek_hashed_entry(seek_key)?;
+                    if let Some((last_next_key, last_next_value)) = self.last_next_result {
+                        if last_next_key == seek_key {
+                            trace!(target: "trie::node_iter", ?seek_key, "skipping redundant hashed seek using last_next_result");
+                            self.current_hashed_entry = Some((last_next_key, last_next_value));
+                            self.last_seeked_hashed_entry = Some(SeekedHashedEntry {
+                                seeked_key: seek_key,
+                                result: self.current_hashed_entry,
+                            });
+                        } else {
+                            trace!(target: "trie::node_iter", ?seek_key, "seeking (last next was different)");
+                            self.current_hashed_entry = self.seek_hashed_entry(seek_key)?;
+                        }
+                        self.last_next_result = None;
+                    } else {
+                        trace!(target: "trie::node_iter", ?seek_key, "seeking (no last next result)");
+                        self.current_hashed_entry = self.seek_hashed_entry(seek_key)?;
+                    }
```

Comment:
this logic looks correct to me, but can we please hide it in the `seek_hashed_entry` method?

Reply:
Done